### PR TITLE
Validate regex

### LINKED ISSUE #362: [Bug]: Invalid regex uncaught exception
### Checklist

- [X] This is an issue for LiveTL, not Hyperchat. I will go to [the Hyperchat repo](https://github.com/LiveTL/HyperChat) to report an issue with Hyperchat.
- [X] This is an issue for the LiveTL web extension, not LiveTL iOS or LiveTL Android. I will go to the [iOS](https://github.com/LiveTL/ios) or [Android](https://github.com/LiveTL/android) repo to report issues for those platforms.
- [X] This issue is not a matter of opinion. If it is, I will use the feature suggestion issue template.
- [X] I have checked for a duplicate or similar issue. I made sure to check closed issues as well.
- [X] I am still able to reproduce the issue after I reinstall the extension.
- [X] I have clear instructions to reproduce the bug, even on another machine.
- [X] I am not suggesting a feature. If I am, I will use the feature suggestion issue template.
- [X] By submitting this issue, you agree to follow our [Code of Conduct](https://github.com/LiveTL/LiveTL/blob/develop/CODE_OF_CONDUCT.md)

### Describe the bug

Typing an invalid regex will cause an uncaught exception, which stops all further JS.

### Steps To Reproduce

1. Create a custom regex filter.
2. Enter `\`
3. Uncaught exception in console.

### Expected behavior

Don't crash when invalid regex is entered.

### What browsers are you seeing the problem on?

Firefox, Chrome, Microsoft Edge

### Anything else?

image

### PR DESCRIPTION
**Pull request by ChrRubin**
Checks that the regex from custom filters are valid, and ignore the those that are invalid. Also includes UI changes to show user that the particular regex is invalid:

image

Fixes #362.

### CODE REVIEW

**Comment by ChrRubin** [comment]
I've also removed auto delete when rule is empty cause its just annoying UX.

File: src/js/filter.js [hunk 1]
```diff
@@ -44,7 +45,7 @@ function userFilter(ufilters, transform = filter => filter) {
   let userRegex = null;
   ufilters.subscribe(filters => {
     userRegex = filters.length
-      ? new RegExp(filters.map(transform).filter(e => e).join('|'))
+      ? validateRegex(filters.map(transform).filter(e => e).join('|'))
```
Comment (r2dev2) [inline src/js/filter.js]: we should make it such that if one regex is wrong, it doesn't disable all the regexs by applying `validateRegex` logic to each regex that is joined

**Review by r2dev2 — CHANGES_REQUESTED** [review]

File: src/js/filter.js (hunk 1 above)
Comment (r2dev2) [reply src/js/filter.js]: I did it in 4836f01

**Review by r2dev2 — APPROVED** [review]

**Comment by ChrRubin** [comment]
Yup those changes seems good (I can't approve my own PR)

**Comment by r2dev2** [comment]
@KentoNishi good for merge?

**Review by KentoNishi — APPROVED** [review]

**Pull request merged**